### PR TITLE
Role/build infra

### PR DESCRIPTION
**Pull request by Polina-Gubina**
Add role dependencies
Add clouds.yaml authentication

### CODE REVIEW

File: build_infrastructure/.yamllint [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
+    max-spaces-after: -1
+    level: error
+  commas:
+    max-spaces-after: -1
+    level: error
+  comments: disable
+  comments-indentation: disable
+  document-start: disable
+  empty-lines:
+    max:
```
Comment (outcatcher) [inline build_infrastructure/.yamllint]: Seems to be a mistake

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
-### Ansible template
-*.retry
-.idea/
-.terraform/
+# Created by .ignore support plugin (hsz.mobi)
+### Ansible template
+*.retry
+.idea/
+.terraform/
+/build_infrastructure/templates/secure.yaml
```
Comment (outcatcher) [inline .gitignore]: better to ignore just `secure.yaml`

File: build_infrastructure/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+galaxy_info:
```
Comment (outcatcher) [inline build_infrastructure/meta/main.yml]: Seems to be info for `darkwizard242.terraform` role, please change to be our own

File: build_infrastructure/tasks/main.yml [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
-
-- name: Terraform build
-  include_tasks: build.yml
+- name: Terraform pre_build
+  include_tasks: pre_build.yml
+
+- name: Terraform build
+  include_tasks: build.yml
+
+- name: Terraform destroy
```
Comment (outcatcher) [inline build_infrastructure/tasks/main.yml]: Infra is always destroyed

File: build_infrastructure/templates/clouds-public.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+clouds:
```
Comment (outcatcher) [inline build_infrastructure/templates/clouds-public.yaml]: `clouds-public.yaml` seems to be redundant here

File: build_infrastructure/molecule/default/molecule.yml [hunk 6]
```diff
@@ -5,38 +5,33 @@ driver:
...
-  name: yamllint
 platforms:
   - name: instance
     image: python:3.7
 provisioner:
   name: ansible
   connection_options:
     ansible_python_interpreter: "PYTHONPATH=$PYTHONPATH:/usr/lib/python3/dist-packages /usr/bin/env python"
+    options:
+      vvv: yes
```
Comment (outcatcher) [inline build_infrastructure/molecule/default/molecule.yml]: Should be disabled

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: build_infrastructure/molecule/default/files/infrastructure/scenario1/variables.tf [hunk 7]
```diff
@@ -0,0 +1,2 @@
+
+
```
Comment (lego963) [inline build_infrastructure/molecule/default/files/infrastructure/scenario1/variables.tf]: Empty file?

File: build_infrastructure/templates/clouds.yaml.j2 [hunk 8]
```diff
@@ -0,0 +1,9 @@
+clouds:
+    infra:
```
Comment (lego963) [inline build_infrastructure/templates/clouds.yaml.j2]: Mb change `infra` to `otc`

File: build_infrastructure/vars/main.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  key: "terraform_state/{{scenario_name}}"
+  endpoint: "obs.eu-de.otc.t-systems.com"
+  bucket: "obs-csm"
+  region: "eu-de"
+
+username: "{{ lookup('env', 'OS_USERNAME') }}"
+password: "{{ lookup('env', 'OS_PASSWORD') }}"
+
+auth_url: https://iam.eu-de.otc.t-systems.com/v3
+user_domain_name: OTC00000000001000000448
```
Comment (lego963) [inline build_infrastructure/vars/main.yml]: Hard link to 448 is it ok?

File: too_simple_server/tasks/main.yml [hunk 10]
```diff
@@ -4,7 +4,7 @@
   register: LATEST_VERSION
 
 - name: Check active version of too_simple_server
-  shell: "{{ server|quote }} --version" # noqa 301 305
+  command: "{{ server|quote }} --version" # noqa 301 305
```
Comment (lego963) [inline too_simple_server/tasks/main.yml]: If we use command we will not use quote filter

**Review by lego963 — CHANGES_REQUESTED** [review]
Add endlines

File: build_infrastructure/vars/main.yml (hunk 9 above)
Comment (Polina-Gubina) [reply build_infrastructure/vars/main.yml]: i have't got any idea... what about you?

File: too_simple_server/tasks/main.yml (hunk 10 above)
Comment (Polina-Gubina) [reply too_simple_server/tasks/main.yml]: shell is't good too, are there other way?

File: build_infrastructure/molecule/default/converge.yml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Converge
+  hosts: all
+  become: yes
+  vars:
+    infra_state: present
+    requirements: 'requirements.txt'
```
Comment (lego963) [inline build_infrastructure/molecule/default/converge.yml]: make double quotes

File: build_infrastructure/tasks/destroy.yml [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+      state: absent
+      force_init: true
+      project_path: "{{ terraform_remote_dir }}/{{ scenario_name }}"
+      backend_config:
+        key: "terraform_state/{{ scenario_name }}"
+        bucket: "obs-csm"
+        region: "eu-de"
+      variables: "{{ tf_variables }}"
+  - name:  Remove terraform directory
+    file: path='/tmp/{{ item }}' state=absent
```
Comment (lego963) [inline build_infrastructure/tasks/destroy.yml]: double quotes

File: build_infrastructure/vars/main.yml (hunk 9 above)
Comment (lego963) [reply build_infrastructure/vars/main.yml]: maybe will we use lookup for project_name and domain name?

File: too_simple_server/tasks/main.yml (hunk 10 above)
Comment (lego963) [reply too_simple_server/tasks/main.yml]: i mean we don't need to use `quote` filter if we moved to command

**Review by lego963 — CHANGES_REQUESTED** [review]
double quotes and endline 1 file more

File: build_infrastructure/molecule/default/molecule.yml [hunk 13]
```diff
@@ -17,26 +15,21 @@ provisioner:
...
-        scenario_name: "infrastructure"
+        scenario_name: "scenario1"
+        terraform_base_dir: "infrastructure"
         key_name: 'test_host_rsa'
         tmp_dir: "/tmp"
-  lint:
-    name: ansible-lint
 scenario:
   test_sequence:
-    - lint
```
Comment (outcatcher) [inline build_infrastructure/molecule/default/molecule.yml:27]: Should be enabled

File: build_infrastructure/tasks/pre_build.yml [hunk 14]
```diff
@@ -1,10 +1,30 @@
...
-      executable: python3
-    register: get_key_output
-    changed_when: get_key_output == 'Private key is up to date\n'
+   - name: Copy Requirements
+     copy:
+       src: "{{ requirements }}"
+       dest: "{{ tmp_dir }}"
+   - name:  Creation openstack directory
+     file:
+       path: /etc/openstack/
```
Comment (outcatcher) [inline build_infrastructure/tasks/pre_build.yml]: better use `~/.config/openstack`

File: build_infrastructure/tasks/main.yml [hunk 15]
```diff
@@ -3,3 +3,8 @@
 
 - name: Terraform build
   include_tasks: build.yml
+  when: desired_state == 'present'
```
Comment (outcatcher) [inline build_infrastructure/tasks/main.yml]: Better name variable `infra_state`

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: build_infrastructure/vars/main.yml (hunk 9 above)
Comment (Polina-Gubina) [reply build_infrastructure/vars/main.yml]: i think only username and password are secure date

**Review by outcatcher — APPROVED** [review]

**Review by lego963 — APPROVED** [review]

**Pull request merged**